codesearch/token/token_test.go

```diff
@@ -67,39 +72,43 @@ func oldTokenize(f io.Reader) []string {
 	return r
 }
 
-// Test generated tokens match old code search
-func TestTrigramTokenizer(t *testing.T) {
-	tt := NewTrigramTokenizer()
-	tt.Reset(bytes.NewReader([]byte(sampleBuf)))
-	newTokens := make([]string, 0)
+// tokenizeBuf applies a tokenizer to a string and returns a sorted slice of
+// tokens.
+func tokenizeBuf(buf string, tt types.Tokenizer) []string {
+	tt.Reset(strings.NewReader(buf))
+	tokens := make([]string, 0)
 	for {
 		tok, err := tt.Next()
 		if err != nil {
 			break
 		}
-		newTokens = append(newTokens, string(tok.Ngram()))
+		tokens = append(tokens, string(tok.Ngram()))
 	}
-	oldTokens := oldTokenize(bytes.NewReader([]byte(sampleBuf)))
+	return tokens
+}
 
-	sort.Strings(oldTokens)
+// Test generated tokens match old code search
+func TestTrigramTokenizer(t *testing.T) {
+	newTokens := tokenizeBuf(sampleBuf, NewTrigramTokenizer())
+	oldTokens := oldTokenize(bytes.NewReader([]byte(sampleBuf)))
 	sort.Strings(newTokens)
+	sort.Strings(oldTokens)
 	assert.Equal(t, oldTokens, newTokens)
 }
 
 func TestWhitespaceTokenizer(t *testing.T) {
-	wt := NewWhitespaceTokenizer()
-	wt.Reset(bytes.NewReader([]byte("this is a string")))
-	tokens := make([]string, 0)
-	for {
-		tok, err := wt.Next()
-		if err != nil {
-			break
-		}
-		tokens = append(tokens, string(tok.Ngram()))
-	}
+	tokens := tokenizeBuf("this is a string", NewWhitespaceTokenizer())
 	assert.Equal(t, []string{"this", "is", "a", "string"}, tokens)
 }
 
+// func TestBuildAllNgramsMatchesTrigrams(t *testing.T) {
```

Comment:
Intentional?

Reply:
yeah gonna fix in a followup -- requires changing the tri tokenizer to stop tokenizing across lines